modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1786 @@
+----
+
+.Error message
+[source, error]
+----
+Node(0) with label `Author` must have the property `name`
+----
+
+======
+
+.Create a relationship that violates an existing relationship property type constraint
+======
+
+
+.Create a `PART_OF` relationship with a `STRING` `order` property, given a property type constraint on the relationship type `PART_OF` restricting the `order` property to `INTEGER` values
+[source, cypher, role=test-fail]
+----
+MATCH (movie:Movie {title:'Iron Man'}), (franchise:Franchise {name:'MCU'})
+CREATE (movie)-[part:PART_OF {order: '1'}]->(franchise)
+----
+
+.Error message
+[source, error]
+----
+Relationship(0) with type `PART_OF` has property `order` of wrong type `String`. Allowed types: INTEGER
+----
+
+======
+
+
+.Create a node that violates an existing node key constraint
+======
+
+.Create an `Actor` node without a `firstname` property, given a node key constraint on `:Actor(firstname, surname)`
+[source, cypher, role=test-fail]
+----
+CREATE (actor:Actor {surname: 'Wood'})
+----
+
+.Error message
+[source, error]
+----
+Node(0) with label `Actor` must have the properties (`firstname`, `surname`)
+----
+
+======
+
+[[removing-an-existing-constrained-property-will-fail]]
+==== Removing existence and key constrained properties will fail
+
+.Remove a node property existence constrained property
+======
+
+.Remove the `name` property from an existing `Author` node, given a property existence constraint on `:Author(name)`
+[source, cypher, role=test-fail]
+----
+MATCH (author:Author {name: 'Virginia Woolf'})
+REMOVE author.name
+----
+
+.Error message
+[source, error]
+----
+Node(0) with label `Author` must have the property `name`
+----
+
+======
+
+
+.Remove a node key constrained property
+======
+
+.Remove the `firstname` property from an existing node `Actor`, given a node key constraint on `:Actor(firstname, surname)`
+[source, cypher, role=test-fail]
+----
+MATCH (actor:Actor {firstname: 'Keanu', surname: 'Reeves'})
+REMOVE actor.firstname
+----
+
+.Error message
+[source, error]
+----
+Node(0) with label `Actor` must have the properties (`firstname`, `surname`)
+----
+
+======
+
+[[modifying-property-constrained-property-will-fail]]
+==== Modifying type constrained properties will fail
+
+.Modify a type constrained property
+======
+
+.Modify the `title` for the `Movie` 'Iron Man' to an `INTEGER` value, given a constraint requiring `title` properties to be of type `STRING`
+[source, cypher, role=test-fail]
+----
+MATCH (m:Movie {title: 'Iron Man'})
+SET m.title = 13
+----
+
+.Error message
+[source, error]
+----
+Node(9) with label `Movie` required the property `title` to be of type `STRING`, but was of type `INTEGER`.
+----
+
+======
+
+
+[[constraint-creation-failures]]
+=== Constraint creation failures
+
+* xref:constraints/managing-constraints.adoc#create-an-already-existing-constraint[]
+* xref:constraints/managing-constraints.adoc#fail-to-create-constraint-due-to-existing-data[]
```

Comment:
I've now added comments to move both other sections out from here, if this is the only one left then maybe it should move in under the section about failing data instead of having it's own here (I assume you don't have more hidden sections :p)

Reply:
fixed in my last commit